testing/test_warnings.py

```diff
@@ -683,3 +683,27 @@ def test_false_function_no_warn(self, testdir):
         self.create_file(testdir, False)
         result = testdir.runpytest()
         result.stdout.fnmatch_lines(["*1 failed in*"])
+
+
+def test_warningschecker_twice(testdir):
```

Comment:
I would go for a simple unittest here:

```python
def test_warningschecker_twice():
    expectation = pytest.warns(UserWarning)
    with expectation:
        warnings.warn("Message A", UserWarning)
    with expectation:
        warnings.warn("Message B", UserWarning)
```

Reply:
@nicoddemus 
Makes sense, sorry for merging it too early.
Will you create a followup?

Reply:
Yep, way simpler! But now it should be another PR?

Reply:
Yep, we can't change this one anymore. Can you do it @Sup3rGeo? If you don't have the time, I will get to it later. 👍

Reply:
Done: https://github.com/pytest-dev/pytest/pull/4622